query-planner-js/src/buildPlan.ts

```diff
@@ -4027,6 +4104,81 @@ function operationForEntitiesFetch(
   return new Operation(subgraphSchema, 'query', entitiesCall, variableDefinitions, operationName);
 }
 
+class Finder {
+  constructor(readonly target: FieldDefinition<any>) {}
+
+  fieldName(): string {
+    return this.target.name;
+  }
+
+  entityType(): NamedType {
+    const entityType = this.target.type;
+    assert(entityType, 'finder must return a defined type');
+    return entityType as NamedType;
+  }
+
+  arguments(): Readonly<ArgumentDefinition<FieldDefinition<any>>[]> {
+    return this.target.arguments();
+  }
+
+  argument(idx: number): { inputType: InputType, name: string } {
+    const arg = this.target.arguments()[idx];
+    assert(arg, `Argument at index ${idx} exists`);
+    assert(arg.type, 'type exists on argument');
+    return {
```

Comment:
Why create some new random structure instead of just returning `arg`? Also, this method is call twice, separately as `finder.argument(0).inputType` and `finder.argument(0).name`, which seem even more unnecessarily wasteful.